lib/routes/routeBackbeat.js

```diff
@@ -0,0 +1,186 @@
+    const body = typeof payload === 'object' ?
+        JSON.stringify(payload) : payload;
+    const httpHeaders = {
+        'x-amz-id-2': log.getSerializedUids(),
+        'x-amz-request-id': log.getSerializedUids(),
+        'content-type': 'application/json',
+        'content-length': Buffer.byteLength(body),
+    };
+    response.writeHead(200, httpHeaders);
+    response.end(body, 'utf8', callback);
+}
+
+function _getRequestPayload(req, cb) {
+    const payload = [];
+    let payloadLen = 0;
+    req.on('data', chunk => {
+        payload.push(chunk);
+        payloadLen += chunk.length;
+    }).on('error', cb)
+    .on('end', () => cb(null, Buffer.concat(payload, payloadLen).toString()));
+}
+
+/*
+PUT /_/backbeat/<bucket name>/<object key>/metadata
+PUT /_/backbeat/<bucket name>/<object key>/data
+*/
+
+function putData(request, response, bucketInfo, objMd, log, callback) {
+    const { bucket, object } = _parseRequest(request);
+
+    const canonicalID = request.headers['x-scal-canonicalId'];
+    const contentMd5 = request.headers['content-md5'];
+    const context = {
+        bucketName: bucket,
+        owner: canonicalID,
+        namespace: NAMESPACE,
+        objectKey: object,
+    };
+    const payloadLen = parseInt(request.headers['content-length'], 10);
+    const backendInfoObj = locationConstraintCheck(
+        request, null, bucketInfo, log);
+    if (backendInfoObj.err) {
+        log.error('error getting backendInfo', {
+            error: backendInfoObj.err,
+            method: 'routeBackbeat',
+        });
+        return callback(errors.InternalError);
+    }
+    const backendInfo = backendInfoObj.backendInfo;
+    return dataStore(
+        context, CIPHER, request, payloadLen, {},
+        backendInfo, log, (err, retrievalInfo, md5) => {
+            if (err) {
+                return callback(err);
+            }
+            if (contentMd5 !== md5) {
+                return callback(errors.BadDigest);
+            }
+            const { key, dataStoreName } = retrievalInfo;
+            const dataRetrievalInfo = [{
+                key,
+                dataStoreName,
+                size: payloadLen,
+                start: 0,
+            }];
+            return _respond(response, dataRetrievalInfo, log, callback);
+        });
+}
+
+function putMetadata(request, response, bucketInfo, objMd, log, callback) {
+    const { bucket, object } = _parseRequest(request);
+
+    return _getRequestPayload(request, (err, payload) => {
+        if (err) {
+            return callback(err);
+        }
+        let omVal;
+        try {
+            omVal = JSON.parse(payload);
+        } catch (err) {
+            // FIXME: add error type MalformedJSON
+            return callback(errors.MalformedPOSTRequest);
+        }
+        // specify both 'versioning' and 'versionId' to create a "new"
+        // version (updating master as well) but with specified
+        // versionId
```

Comment:
afaik this shouldn't work with real metadata/arsenal, the versioning request processor will return `BadRequest`:
https://github.com/scality/Arsenal/blob/master/lib/versioning/VersioningRequestProcessor.js#L264

Were you planning to modify the vrp code to accomplish what you want?